### ISSUE TITLE
MultiNeuralNetConfLayerBuilderTest::testRbmSetup - "no gradient with key found"

### ISSUE THREAD

**Issue by EronWright** [deeplearning4j/deeplearning4j#566]
```
testRbmSetup(org.deeplearning4j.nn.conf.MultiNeuralNetConfLayerBuilderTest)  Time elapsed: 3.283 sec  <<< ERROR!
java.lang.IllegalStateException: Illegal key 0_vb no gradient with key found
	at org.deeplearning4j.nn.gradient.DefaultGradient.gradient(DefaultGradient.java:48)
	at org.deeplearning4j.optimize.solvers.BaseOptimizer.optimize(BaseOptimizer.java:138)
	at org.deeplearning4j.optimize.Solver.optimize(Solver.java:52)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(MultiLayerNetwork.java:1196)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(MultiLayerNetwork.java:1225)
	at org.deeplearning4j.nn.conf.MultiNeuralNetConfLayerBuilderTest.testRbmSetup(MultiNeuralNetConfLayerBuilderTest.java:141)
```

**Comment by nyghtowl** [comment]
RBM works for pretrain/finetune when backprop is false.

RBM layer needs backpropGradient implemented to account for visible bias layer. Currently only updating hidden bias when included in MultiLayerNetwork and backprop true. Also appears computeGradientAndScore should be reviewed and updated for backprop condition.

**Comment by agibsonccc** [comment]
https://github.com/deeplearning4j/deeplearning4j/commit/78eb0c7d6655e348bce6a26131bc164880633a15
